src/main/scala/edu/ie3/mobsim/model/ChargingBehavior.scala

```diff
@@ -74,39 +75,49 @@ object ChargingBehavior extends LazyLogging {
         logger.debug(
           s"${ev.getId} arrives at charging hub and wants to start charging."
         )
-        ev.getDestinationPoi.nearestChargingStations.keys.headOption
-          .map(_.getUuid)
+        (
+          ev.destinationPoi.nearestChargingStations.keys.headOption
+            .map(_.uuid),
+          None
+        )
       } else {
         /* Update charging prices memory of EV to have a reference for the prices of specific charging stations */
-        ev.getDestinationPoi.nearestChargingStations.keys.foreach { cs =>
+        val priceQueue: mutable.Queue[Double] = mutable.Queue.empty
+
+        ev.destinationPoi.nearestChargingStations.keys.foreach { cs =>
           currentPricesAtChargingStations
-            .get(cs.getUuid)
-            .map(ev.updateChargingPricesMemory)
+            .get(cs.uuid)
+            .map(priceQueue += _)
         }
+        val evWithUpdatedPriceMemory = ev.updateChargingPricesMemory(priceQueue)
 
         /* If EV wants to charge, rank available charging stations and choose the best */
         val evWantsToCharge = doesEvWantToCharge(ev, seed)
         if (evWantsToCharge) {
           /* Create rating for each possible charging station */
-          createRatingsForChargingStations(
+          val ratingMap = createRatingsForChargingStations(
             ev,
             currentPricesAtChargingStations,
             currentlyAvailableChargingPoints,
             maxDistance
-          ).maxByOption(_._2).map(_._1)
+          )
+          (
+            ratingMap.maxByOption(_._2).map(_._1),
```

Comment:
Please don't use the shorthand tuple methods `_1` and `_2`, but rather go for a case-mapping like this:

```suggestion
            ratingMap
              .maxByOption { case (_, rating) => rating }
              .map { case (cs, _) => cs },
```

Yes, this way we have longer code, but it's actually easier to grasp on first sight what is happening.